src/services/download/gcs.rs

```diff
@@ -156,56 +145,20 @@ pub async fn download_source(
     let token = get_token(&source.source_key).await?;
     log::debug!("Got valid GCS token: {:?}", &token);
 
-    let url = format!(
+    let url = Url::parse(&format!(
         "https://www.googleapis.com/download/storage/v1/b/{}/o/{}?alt=media",
         percent_encode(source.bucket.as_bytes(), PATH_SEGMENT_ENCODE_SET),
         percent_encode(key.as_bytes(), PATH_SEGMENT_ENCODE_SET),
-    );
+    ))
+    .unwrap();
 
-    let mut backoff = ExponentialBackoff::from_millis(10).map(jitter).take(3);
-    let response = loop {
-        let result = start_request(&url, &token).await;
-        match backoff.next() {
-            Some(duration) if result.is_err() => delay(duration).await,
-            _ => break result,
-        }
-    };
+    let mut headers = BTreeMap::new();
+    headers.insert(
+        "authorization".to_owned(),
+        format!("Bearer {}", token.access_token),
+    );
 
-    match response {
-        Ok(response) => {
-            if response.status().is_success() {
-                log::trace!("Success hitting GCS {} (from {})", &key, source.bucket);
-                let stream = response
-                    .payload()
-                    .compat()
-                    .map(|i| i.map_err(|e| e.context(DownloadErrorKind::Io).into()));
-                super::download_stream(
-                    SourceFileId::Gcs(source, download_path),
-                    stream,
-                    destination,
-                )
-                .await
-            } else {
-                log::trace!(
-                    "Unexpected status code from GCS {} (from {}): {}",
-                    &key,
-                    source.bucket,
-                    response.status()
-                );
-                Ok(DownloadStatus::NotFound)
-            }
-        }
-        Err(e) => {
-            log::trace!(
-                "Skipping response from GCS {} (from {}): {} ({:?})",
-                &key,
-                source.bucket,
-                &e,
-                &e
-            );
-            Ok(DownloadStatus::NotFound)
-        }
-    }
+    super::http::download_url(url, &headers, destination).await
```

Comment:
This didn't used to follow redirects, now it does.  I'm not sure how much we care, but it is a change.